lib/fluent/plugin/mongo_auth.rb

```diff
@@ -8,23 +8,25 @@ def self.included(klass)
         config_param :password, :string, default: nil, secret: true
         desc "MongoDB authentication database"
         config_param :auth_source, :string, default: nil
+        desc "MongoDB authentication mechanism"
+        config_param :auth_mech, :string, default: nil
       }
     end
   end
 
   module MongoAuth
     def authenticate(client)
-      unless @user.nil? || @password.nil?
-        begin
-          if @auth_source.nil?
-            client = client.with(user: @user, password: @password)
-          else
-            client = client.with(user: @user, password: @password, auth_source: @auth_source)
-          end
-        rescue Mongo::Auth::Unauthorized => e
-          log.fatal e
-          exit!
+      begin
+        if [@user, @password, @auth_source].all?
+          client = client.with(user: @user, password: @password, auth_source: @auth_source)
+        elsif [@user, @password].all?
+          client = client.with(user: @user, password: @password)
+        elsif [@user, @auth_source, @auth_mech].all?
```

Comment:
Any chance to add test case for newly added lines?

Reply:
Good point, will do.